scripts/check-docs/src/main.rs

```diff
@@ -0,0 +1,329 @@
+                let anchor_file = PathBuf::from(&capture[3]);
+                let anchor_name = capture[4].to_owned();
+
+                let the_path = include_file.parent().unwrap().join(anchor_file);
+
+                let anchor_file = the_path.canonicalize().unwrap_or_else(|err| {
+                    panic!("{the_path:?} when canonicalized gives error {:?}", err)
+                });
+
+                Ok(Include {
+                    anchor_name,
+                    anchor_file,
+                    include_file,
+                    line_no,
+                })
+            })
+            .collect::<Result<Vec<_>, Error>>()
+    };
+
+    apply_regex(Regex::new(
+        r"^(\S+):(\d+):\s*\{\{\s*#include\s*(\S+)\s*:\s*(\S+)\s*\}\}",
+    )?)
+}
+
+fn filter_valid_anchors(starts: Vec<Anchor>, ends: Vec<Anchor>) -> (Vec<Anchor>, Vec<Error>) {
+    let find_anchor_end_by_name = |anchor_name: &str, file: &Path| {
+        ends.iter()
+            .filter(|el| el.name == *anchor_name && el.file == file)
+            .collect::<Vec<_>>()
+    };
+
+    let (pairs, errors):(Vec<_>, Vec<_>) = starts.into_iter().map(|start| {
+        let matches_by_name = find_anchor_end_by_name(&start.name, &start.file);
+
+        let (begin, end) = match matches_by_name.as_slice() {
+            [single_match] => Ok((start, (*single_match).clone())),
+            [] => Err(anyhow!("Couldn't find a matching end anchor for {start:?}")),
+            multiple_ends => Err(anyhow!("Found too many matching anchor ends for anchor: {start:?}. The matching ends are: {multiple_ends:?}")),
+        }?;
+
+        match check_validity_of_anchor_pair(&begin, &end) {
+            None => Ok((begin, end)),
+            Some(err) => {
+                let err_msg = err.to_string();
+                Err(anyhow!("{err_msg}"))
+            }
+        }
+    }).partition_result();
+
+    let additional_errors = filter_unused_ends(&ends, &pairs)
+        .into_iter()
+        .map(|unused_end| anyhow!("Missing anchor start for {unused_end:?}"))
+        .collect::<Vec<_>>();
+
+    let start_only = pairs.into_iter().map(|(begin, _)| begin).collect();
+
+    (start_only, chain!(errors, additional_errors).collect())
+}
+
+fn filter_unused_ends<'a>(ends: &'a [Anchor], pairs: &[(Anchor, Anchor)]) -> Vec<&'a Anchor> {
+    ends.iter()
+        .filter(|end| {
+            let end_used_in_pairs = pairs.iter().any(|(_, used_end)| *end == used_end);
+            !end_used_in_pairs
+        })
+        .collect()
+}
+
+fn check_validity_of_anchor_pair(begin: &Anchor, end: &Anchor) -> Option<anyhow::Error> {
+    if begin.line_no > end.line_no {
+        Some(anyhow!("The end of the anchor appears before the beginning. End anchor: {end:?}. Begin anchor: {begin:?}"))
+    } else {
+        None
+    }
+}
+
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+struct Anchor {
+    line_no: usize,
+    name: String,
+    file: PathBuf,
+}
+
+#[allow(dead_code)]
+enum TestEnum {
```

Comment:
You can put this inside the code guarded by the `cfg` for it not to be dead in production code.